deals/deals.go

```diff
@@ -75,20 +73,28 @@ func (m *Module) Store(ctx context.Context, waddr string, data io.Reader, dcfgs
 	}()
 	var size int64
 	if size, err = io.Copy(f, data); err != nil {
-		return cid.Undef, nil, fmt.Errorf("error when copying data to tmpfile: %s", err)
+		return cid.Undef, 0, fmt.Errorf("error when copying data to tmpfile: %s", err)
 	}
-	gbSize := float64(size) / float64(1<<30)
 	ref := api.FileRef{
 		Path:  f.Name(),
 		IsCAR: isCAR,
 	}
 	dataCid, err := m.api.ClientImport(ctx, ref)
 	if err != nil {
-		return cid.Undef, nil, fmt.Errorf("error when importing data: %s", err)
+		return cid.Undef, 0, fmt.Errorf("error when importing data: %s", err)
 	}
+	return dataCid, size, nil
```

Comment:
Ok, just want to make sure I understand... `Import` calls the Lotus client `ClientImport`, passing the file name and `isCar`. Lotus `ClientImport` adds the data to the Lotus node's configured IPFS node? Is the data simply added, or pinned as well?

Reply:
AFAICT, I think it would be pinned via `Store`, this just makes it available

Reply:
@asutula , @sanderpick , yes. I'll extend a bit the explanation just in case.

`deals.Import()` import data to the Lotus node by sending it from Powergate. This make Lotus have the available data in a further call to `deals.Store`.

Now, something important. The FFS world *never* calls `deals.Import`, since in FFS the Lotus client is directly connected to IPFS. So it doesn't need to call `deals.Import` since the Lotus node has the data _already imported_ in IPFS.

The `deals.Import()` API is mostly useful for other use cases that uses the Deal Module which doesn't rely on a Lotus node connected to IPFS. In that case, the client should provide de deal data before making the deal.

Makes sense?

Reply:
Yep, thank you